File: scratch/ws_client.js
const WebSocket = require('ws');
const ws = new WebSocket('ws://localhost:8080/ws-test-path?query=abc');

ws.on('open', function open() {
  console.log('Connected!');
  ws.send('Hello from Node');
});

ws.on('message', function incoming(data) {
  console.log('Received: %s', data);
  ws.close();
});

ws.on('error', function error(err) {
  console.error('Error: %s', err);
});

File: scratch/ws_client.py
import asyncio
import websockets

async def test_ws():
    uri = "ws://localhost:8080/ws-test-path?query=123"
    async with websockets.connect(uri) as websocket:
        await websocket.send("Hello Wolf!")
        response = await websocket.recv()
        print(f"Server replied: {response}")

if __name__ == "__main__":
    asyncio.run(test_ws())
